sw/device/lib/dif/dif_otbn.h

```diff
@@ -0,0 +1,167 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_OTBN_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_OTBN_H_
+
+#include "sw/device/lib/base/mmio.h"
+#include <stddef.h>
+#include <stdint.h>
+
+// Header Extern Guard (so header can be used from C and C++)
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * Width of wide instructions in OTBN, in bit.
+ */
+#define OTBN_WLEN 256
+
+/**
+ * Configuration for initializing an OTBN device.
+ */
+typedef struct dif_otbn_config { mmio_region_t base_addr; } dif_otbn_config_t;
+
+/**
+ * Internal state of a OTBN device.
+ *
+ * Instances of this struct must be initialized by `dif_otbn_init()` before
+ * being passed to other functions.
+ */
+typedef struct dif_otbn { mmio_region_t base_addr; } dif_otbn_t;
+
+/**
+ * Generic return codes for the functions in this library.
+ */
+typedef enum dif_otbn_result {
+  /**
+   * The function succeeded.
+   */
+  kDifOtbnOk,
+  /**
+   * The function failed a non-specific assertion.
+   *
+   * This error is not recoverable.
+   */
+  kDifOtbnError,
+  /**
+   * There is a problem with the argument(s) to the function.
+   *
+   * This return code is only returned before the function has any side effects.
+   *
+   * This error is recoverable and the operation can be retried after correcting
+   * the problem with the argument(s).
+   */
+  kDifOtbnBadArg,
+
+} dif_otbn_result_t;
+
+/**
+ * Initialize a OTBN device using `config` and return its internal state.
+ *
+ * A particular OTBN device must first be initialized by this function
+ * before calling other functions of this library.
+ *
+ * @param config Configuration for initializing an OTBN device.
+ * @param otbn OTBN instance that will store the internal state of the
+ *             initialized OTBN device.
+ * @return `kDifOtbnBadArg` if `config` or `otbn` is `NULL`,
+ *         `kDifOtbnOk` otherwise.
```

Comment:
This is already implicit. Most DIFs simply say "The result of the operation" here, instead.